### PR TITLE
Add share button for posts

### LINKED ISSUE #2117: adding 'Share' button
You read blogs and you might want to share that blog post or bookmark it (no way to do it at the moment (one page) unless you take a screenshot or use 'search' to find it again) 

- What to share? URL of original post. 
- Button location? at the end of a blog post or post's author info section.
- How it looks like? Not a feature to be highlighted, so only one button displayed and sharing options could be in a dialog after  a click or a dropdown on hovering, example share button on iOS:
download

(This could be for future student(s) who is interested)

### PR DESCRIPTION
**Pull request by tuanthanh2067**
## Issue This PR Addresses

Fixes #2117 

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [X] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Checklist

- [ ] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Review by humphd — CHANGES_REQUESTED** [review]
This is very cool!  I think the icon isn't right, since it looks like it's going to take me some where.  Should we use something that's more common for copying (two overlayed squares)?  When I hover over it, I should get a popup that says I can copy the link if I click.  When I click, the "copied" popup is too small.

There are eslint issues to fix.

**Review by humphd — CHANGES_REQUESTED** [review]
This is way better, nice job.  One thing to fix still: the baseline of the text and the icon don't match:

Screen Shot 2021-11-30 at 7 31 56 PM

**Comment by tuanthanh2067** [comment]
I'm not sure how to fix the baseline thing, any suggestion?

File: src/web/src/components/Posts/ShareButton.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    navigator.clipboard.writeText(url);
+    setIsCopiedToClipboard(true);
+
+    setTimeout(() => {
+      setIsCopiedToClipboard(false);
+    }, 3000);
+  };
+
+  return !isCopiedToClipboard ? (
+    <ButtonTooltip title="Copy url" arrow placement="top" TransitionComponent={Zoom}>
```
Comment (humphd) [inline src/web/src/components/Posts/ShareButton.tsx:46]: `Copy URL` (uppercase on URL)

**Review by humphd — CHANGES_REQUESTED** [review]
I tried to fix the baseline in dev tools, and can't really make it much better.  This icon doesn't want to play nicely with adjacent text.  Let's go with it and maybe someone will solve it later.

One small text fix, and please squash + rebase.

**Review by humphd — APPROVED** [review]
❤️

**Review by HyperTHD — APPROVED** [review]

**Comment by humphd** [comment]
Man, we need to do a release to get all this good stuff in production!  Great work @tuanthanh2067!

**Pull request merged**